source/SH3/graphics/texture.cpp

```diff
@@ -9,57 +9,383 @@
+        // that the whole palette occupies ~ paletteSize/entrySize bytes of space, contains a total of
+        // entrySize/bypp colors per block, which therefore means we have a total of nBlocks * col_per_block colors,
+        // which equates to about 256-colors in total (which seems accurate for an 8-bit texture).
+
+        const std::size_t nBlocks = (pal_header.paletteSize / pal_header.entrySize) / pal_header.bytes_per_pixel;
+        const std::size_t colorsPerBlock = pal_header.entrySize / pal_header.bytes_per_pixel;
+
+        palette.resize(colorsPerBlock * nBlocks);
+
+        for(std::size_t block = 0; block < nBlocks; ++block)
+        {
+            std::size_t read = file.ReadData(static_cast<void*>(&palette[block * colorsPerBlock]), pal_header.entrySize, e);
+
+            if(read != pal_header.entrySize)
+            {
+                Log(LogLevel::WARN, "sh3_texture::Load( ): Warning: Number of bytes read in palette block != pal_header.entrySize! (Expected %d, got %d)", read, pal_header.entrySize);
+                break;
+            }
+
+            // Each block is 256 bytes.
+            // Seek to next block.
+            file.Seek(256 - pal_header.entrySize, std::ios_base::cur);
+        }
+
+        // we need to swap some colors every 32 pixels, starting from the 8th
+        static constexpr std::size_t swapDistance = 32; // distance between swaps
+        static constexpr std::size_t swapSize = 8; // amount of colors to swap
+
+        if(palette.size() > 8)
+        {
+            for(auto iter = next(begin(palette), 8); distance(iter, end(palette)) > swapDistance; advance(iter, swapDistance))
+            {
+                // swap 8 colors
+                const auto swapBlock = next(iter, swapSize);
+                if(distance(swapBlock, end(palette)) < swapSize)
+                {
+                    Log(LogLevel::WARN, "Palette doesn't have enough colors left for swapping.");
+                    break;
+                }
+                std::swap_ranges(iter, swapBlock, swapBlock);
+            }
+        }
+
+
+        // Now that we've completely loaded the palette in its entirety, we can get the 8-bit index value
+        // from the data section of the file and get it's color in the palette!
+
+        //===---THIS IS A CLUSTER FUCK FOR NOW UNTIL WE UNDERSTAND HOW IN THE NAME OF CHRIST THIS WORKS---===//
+        std::vector<std::uint8_t> iBuffer;  // Our index buffer that we put transformed indecies into
+
+        data.resize(header.texWidth * header.texHeight * 3); // We strip the Alpha channel from the BGRA pixel beacuse it is hard locked to 0x80 (not 0xFF!!)
+        iBuffer.resize(header.texSize);
+
+        file.Seek(offset + (header.texFileSize - header.texSize), std::ios_base::beg); // Seek to the beginning of data
+
+        if(header.texWidth > 96) // Apparently this is the distortion flag?!?!
+        {
+            if(header.texWidth % 16 != 0)
+            {
+
```

Comment:
> Yeah, I honestly doubt we'll run into this, otherwise SH3 wouldn't have worked in the first place haha.

Yah, I'm overly cautious about making assumptions.
If the data is corrupted it's highly probable other errors will make it fail anyways. Failing gracefully is pretty, but not really useful in this case.
And I doubt there's any mods that tried to mess with these paletted textures.